scrunch/datasets.py

```diff
@@ -127,22 +130,20 @@ def get_dataset(dataset, connection=None, editor=False, project=None):
                 "Authenticate first with scrunch.connect() or by providing "
                 "config/environment variables")
     root = connection
-    if project:
-        if isinstance(project, six.string_types):
-            project_obj = get_project(project, connection)
-            ds = project_obj.get_dataset(dataset)
-        else:
-            ds = project.get_dataset(dataset)
-    else:
+
+    try:
+        shoji_ds = root.datasets.by('name')[dataset].entity
+    except KeyError:
         try:
-            shoji_ds = root.datasets.by('name')[dataset].entity
+            shoji_ds = root.datasets.by('id')[dataset].entity
         except KeyError:
             try:
-                shoji_ds = root.datasets.by('id')[dataset].entity
-            except KeyError:
+                dataset_url = '{}datasets/{}/'.format(root.self, dataset)
```

Comment:
Better even, you should be able to do `root.catalogs.datasets` and only need to add the dataset id. That makes the URL guessing even smaller.
For URLs better to use something like `urljoin(root.catalogs.datasets, '{}/'.format(dataset))`